File: Difficulty: Medium/Generate all binary strings/generate-all-binary-strings.cpp
class Solution {
  public:
    vector<string> binstr(int n) {
        // code here
        vector<string> res;

        for (int i = 0; i < (1 << n); i++) {
            string s;
    
            // build string from bits of i
            for (int j = n - 1; j >= 0; --j)
                s += ((i >> j) & 1) ? '1' : '0';
    
            res.push_back(s);
        }
        
        return res;
    }
};

File: Difficulty: Medium/Generate all binary strings/generate-all-binary-strings.java
class Solution {
    public ArrayList<String> binstr(int n) {
        // code here
        ArrayList<String> list = new ArrayList<>();
        generate(n,"",list);
        return list;   
    }
    
    public static void generate(int n, String curr, ArrayList<String> result){
        
        if(curr.length() == n){
            result.add(curr);
            return;
        }
        
        generate(n, curr + "0",result);
        generate(n, curr + "1", result);
    }
}
